src/engraving/rw/read302/read302.cpp

```diff
@@ -248,6 +251,159 @@ bool Read302::readScore302(Score* score, XmlReader& e, ReadContext& ctx)
         CompatUtils::assignInitialPartToExcerpts(score->masterScore()->excerpts());
     }
 
+    for (Part* part : score->parts()) {
```

Comment:
I think it would be preferable to move this code into a separate method, because right now the `readScore302` method is becoming very long and it isn't immediately clear what this enormous `for` loop is doing (which would be solved by putting it in a method with a descriptive name).